src/javaserver/Handlers/ConnectionHandler.java

```diff
@@ -13,10 +14,12 @@
 
     private PrintWriter writer;
     private Socket socket;
+    private Logger logger;
     private BufferedReader reader;
 
-    public ConnectionHandler(Socket socket) throws Exception {
+    public ConnectionHandler(Socket socket, Logger logger) throws Exception {
```

Comment:
This is not new code, but the condition in the `while` statement should probably be extracted into another method. I would also avoid using single line `while`/`if` statements without curly braces.